torchgeo/datasets/sentinel.py

```diff
@@ -97,9 +97,9 @@ def __init__(
         Raises:
             FileNotFoundError: if no files are found in ``root``
         """
-        self.bands = bands if bands else self.all_bands
+        bands = bands or self.all_bands
```

Comment:
So `band_indices` is `Optional[Sequence[str]]` and `bands` is `Sequence[str]`. And `band_indices = None` and `bands = []` mean the same thing? I wonder if it would be more clear to match these such that `bands = None` means use all bands.

Reply:
Updated `bands` to be `Optional[Sequence[str]]` where setting `bands=None` defaults to all_bands